source/index.ts

```diff
@@ -553,7 +564,7 @@ export const assert: Assert = {
 	boolean: (value: unknown): asserts value is boolean => assertType(is.boolean(value), 'boolean', value),
 	symbol: (value: unknown): asserts value is symbol => assertType(is.symbol(value), 'symbol', value),
 	numericString: (value: unknown): asserts value is string => assertType(is.numericString(value), AssertionTypeDescription.numericString, value),
-	array: <T = unknown>(value: unknown): asserts value is T[] => assertType(is.array(value), 'Array', value),
+	array: <T = unknown>(value: unknown, assertion?: (element: unknown) => asserts element is T): asserts value is T[] => assertType(is.array(value, assertion), 'Array', value),
```

Comment:
I don't think that `is.array` and `assert.array` are compatible.

We have two options:

1. The `assertion` parameter should be a `type guard` for both `is.array` and `assert.array`, and it would work.
2. You'd keep the `assertion` parameter as an `asserts ...` and the `assert.array` implementation would be less simple than the rest of the methods.

Reply:
Also, as we talked, a better API would be to use `type guards` in `is.array`, since we don't want to `try-catch` the assertion, as we might end up ignoring actual exceptions thrown by user code.